.github/workflows/my-pipeline.yml

```diff
@@ -0,0 +1,42 @@
+name: Continuous Integration
+on: 
+  push:
+    paths-ignore:
+      - "README.md"
+      - ".gitignore"
+  pull_request:
+    paths-ignore:
+      - "README.md"
+      - ".gitignore"
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build and Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: docker build --target test --tag my-test-image .
+      - run: docker run --env-file .env.test my-test-image:latest
+  docker-push-and-deploy:
+    name: Push to Docker Hub and Deploy
+    runs-on: ubuntu-latest
+    needs: build
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and tag images
+        run: |-
+          docker build --target production --tag confuzzled/my-todo-app:latest .
+          docker tag confuzzled/my-todo-app:latest confuzzled/my-todo-app:$GITHUB_SHA
+      - name: Push images
+        run: |-
+          docker push confuzzled/my-todo-app:latest
+          docker push confuzzled/my-todo-app:$GITHUB_SHA
+      - name: Trigger deploy webhook
+        run: curl -dH -X POST '${{ secrets.DEPLOY_WEBHOOK_URL }}'
```

Comment:
One nice addition here can be adding the [--fail](https://curl.se/docs/manpage.html#-f) flag, as otherwise curl reports success as long as a successful connection was made, even if the response reports a failure (e.g. a status code > 400) which can lead to confusing pipeline results